tests/maintenance/test_maintenance.py

```diff
@@ -92,3 +92,12 @@ def test_report_add_remove():
 
     assert len(report.entries) == 1
     assert report.last_updated_by == "jazhang"
+
+
+def test_export_empty_report():
+    # create an empty report with out data passed
+    report = MaintenanceReport()
+    assert report.last_updated is None
+
+    # export empty report shouldn't have problem
+    report._export_dict()
```

Comment:
may validate the last_updated as current date as well

Reply:
I didn't remember how the timestamp was tested..., in this case, what I can think of is mock the datetime.datetime.utcnow() with a fake timestamp and check it, but doesn't make much sense to me. Do you think this is good or just a waste of lines? any other test methods?

Reply:
I guess I can check the output timestamp, see if it's a timestamp with right format.

Reply:
Checking the timestamp to be the current time/mocked time in case of an empty report would have been better. But looking at the bigger picture, adding the timestamp is to ensure the report doesn't crash. So checking if timestamp exist should be fine too.

Reply:
```
current_date = datetime.datetime.now()
    with patch.object(datetime, "datetime", Mock(wraps=datetime.datetime)) as mocked_datetime:
        mocked_datetime.now.return_value = current_date
        report = MaintenanceReport()._export_dict()

    assert report.get('last_updated') == datetime.datetime.strftime(current_date, "%Y-%m-%dT%H:%M:%SZ")
```